src/test/functional/cypress/integration/feature/admin/membership.spec.js

```diff
@@ -0,0 +1,113 @@
+describe('Admin - Membership', function () {
+  beforeEach(function () {
+    cy.task('db:reset');
+    cy.fixture('chars').as('strings');
+    cy.intercept('GET', '**/api/v2/admin/memberships?*').as('getMembership');
+    cy.intercept('POST', '**/api/v2/admin/memberships').as('postMembership');
+    cy.intercept('PUT', '**/api/v2/admin/memberships/*').as('updateMembership');
+    cy.intercept('DELETE', '**/api/v2/admin/memberships').as(
+      'deleteMembership',
+    );
+    cy.fixture('user').then(({admin}) => {
+      this.user = admin;
+    });
+  });
+
+  // Read
+  describe('List Memberships', function () {
+    it('Membership list is loaded', function () {
+      cy.loginTo(this.user, '/admin/membership');
+      cy.wait('@getMembership');
+      cy.toast('info', 'No Records Found');
+    });
+  });
+
+  // Create
+  describe('create membership', function () {
+    it('add membership', function () {
+      cy.loginTo(this.user, '/admin/saveMemberships');
+      cy.getOXD('form').within(() => {
+        cy.getOXDInput('Name').type(this.strings.chars30.text);
+        cy.getOXD('button').contains('Save').click();
+      });
+      cy.wait('@postMembership');
+      cy.toast('success', 'Successfully Saved');
+      cy.task('db:snapshot', {name: 'Membership'});
+    });
+
+    it('checking length validation and required validation on membership', function () {
+      cy.task('db:restore', {name: 'Membership'});
+      cy.loginTo(this.user, '/admin/saveMemberships');
+      cy.getOXD('form').within(() => {
+        cy.getOXDInput('Name')
+          .setValue(this.strings.chars51.text)
+          .isInvalid('Should not exceed 50 characters');
+        cy.getOXDInput('Name').setValue('').isInvalid('Required');
+        cy.getOXDInput('Name')
+          .setValue(this.strings.chars30.text)
+          .isInvalid('Already exists');
+      });
+    });
+
+    it('cancel button behaviour on add membership form', function () {
+      cy.loginTo(this.user, '/admin/saveMemberships');
+      cy.getOXD('form').within(() => {
+        cy.getOXDInput('Name').type(this.strings.chars30.text);
+        cy.getOXD('button').contains('Cancel').click();
+      });
+      cy.wait('@getMembership');
+      cy.getOXD('pageTitle').contains('Memberships');
+    });
+  });
+
+  //Update
+  describe('Update Memberships', function () {
+    it('Edit Memberships', function () {
+      cy.task('db:restore', {name: 'Membership'});
+      cy.loginTo(this.user, '/admin/saveMemberships/1');
+      cy.getOXD('form').within(() => {
+        cy.getOXDInput('Name').clear().type('Updated Name');
+        cy.getOXD('button').contains('Save').click();
+      });
+      cy.wait('@updateMembership');
+      cy.toast('success', 'Successfully Updated');
+      // eslint-disable-next-line cypress/no-unnecessary-waiting
+      cy.wait(3000);
```

Comment:
wait is said to be an anti-pattern in cypress. 🤕

Reply:
But if i remove that one, testcase will fail sometimes. face_with_head_bandage